src/engine/Graphics/NineSlice.ts

```diff
@@ -0,0 +1,559 @@
+      this.config.sourceConfig.topMargin,
+      this.canvasE.width,
+      this.canvasE.height,
+      0,
+      0,
+      this.canvasE.width,
+      this.canvasE.height
+    );
+
+    //middle right slice
+    this.canvasF.width = this.config.sourceConfig.rightMargin;
+    this.canvasF.height = this.config.sourceConfig.height - this.config.sourceConfig.topMargin - this.config.sourceConfig.bottomMargin;
+    const Ftx = this.canvasF.getContext('2d');
+    Ftx?.drawImage(
+      this.sourceSprite,
+
+      this.config.sourceConfig.width - this.config.sourceConfig.rightMargin,
+      this.config.sourceConfig.topMargin,
+      this.canvasF.width,
+      this.canvasF.height,
+      0,
+      0,
+      this.canvasF.width,
+      this.canvasF.height
+    );
+
+    //bottom left slice
+    this.canvasG.width = this.config.sourceConfig.leftMargin;
+    this.canvasG.height = this.config.sourceConfig.bottomMargin;
+    const Gtx = this.canvasG.getContext('2d');
+    Gtx?.drawImage(
+      this.sourceSprite,
+      0,
+      this.config.sourceConfig.height - this.config.sourceConfig.bottomMargin,
+      this.canvasG.width,
+      this.canvasG.height,
+      0,
+      0,
+      this.canvasG.width,
+      this.canvasG.height
+    );
+
+    //bottom slice
+    this.canvasH.width = this.config.sourceConfig.width - this.config.sourceConfig.leftMargin - this.config.sourceConfig.rightMargin;
+    this.canvasH.height = this.config.sourceConfig.bottomMargin;
+    const Htx = this.canvasH.getContext('2d');
+    Htx?.drawImage(
+      this.sourceSprite,
+      this.config.sourceConfig.leftMargin,
+      this.config.sourceConfig.height - this.config.sourceConfig.bottomMargin,
+      this.canvasH.width,
+      this.canvasH.height,
+      0,
+      0,
+      this.canvasH.width,
+      this.canvasH.height
+    );
+
+    //bottom right slice
+    this.canvasI.width = this.config.sourceConfig.rightMargin;
+    this.canvasI.height = this.config.sourceConfig.bottomMargin;
+    const Itx = this.canvasI.getContext('2d');
+    Itx?.drawImage(
+      this.sourceSprite,
+      this.sourceSprite.width - this.config.sourceConfig.rightMargin,
+      this.config.sourceConfig.height - this.config.sourceConfig.bottomMargin,
+      this.canvasI.width,
+      this.canvasI.height,
+      0,
+      0,
+      this.canvasI.width,
+      this.canvasI.height
+    );
+  }
+
+  /**
+   * Clones the 9 slice
+   */
+
+  clone(): NineSlice {
+    return new NineSlice(this.config);
+  }
+
+  /**
+   * Returns the number of tiles
+   */
+
+  _getNumberOfTiles(tilesize: number, destinationSize: number, strategy: NineSliceStretch): number {
+    switch (strategy) {
+      case NineSliceStretch.Stretch:
+        return 1;
+      case NineSliceStretch.Tile:
+        return Math.ceil(destinationSize / tilesize);
+      case NineSliceStretch.TileFit:
+        return Math.ceil(destinationSize / tilesize);
+    }
+  }
+
+  /**
+   * Returns the position and size of the tile
+   */
+  _calculateParams(
```

Comment:
Should this method be part of  the public api?

Reply:
not based on my implementation

Reply:
in that case, the method should get a proper access modifier like private or protected, just prefixing it with underscore indicates that intention already but you're not enforcing it, without a modifier TS still leaves it as public

Reply:
any reason that should impact functionality?

Reply:
fixed